platform/vechain/api.go

```diff
@@ -0,0 +1,119 @@
+package vechain
+
+import(
+	"github.com/gin-gonic/gin"
+	"github.com/trustwallet/blockatlas/util"
+	"github.com/trustwallet/blockatlas/models"
+	"github.com/trustwallet/blockatlas/coin"
+	"github.com/spf13/viper"
+	"net/http"
+	"math/big"
+	"strings"
+	"strconv"
+	"sync"
+)
+
+var client = Client{
+	HTTPClient: http.DefaultClient,
+}
+
+var wg sync.WaitGroup
+
+func Setup(router gin.IRouter) {
+	router.Use(util.RequireConfig("vechain.api"))
+	router.Use(func(c *gin.Context) {
+		client.URL = viper.GetString("vechain.api")
+		c.Next()
+	})
+	router.GET("/:address", getTransactions)
+}
+
+func getTransactions(c *gin.Context) {
+	addr := strings.ToLower(c.Param("address"))
+	txs, _ := client.GetAddressTransactions(addr)
+
+	var receiptsMap = make(map[string]TxReceipt)
+	receiptsChan := make(chan TxReceipt, len(txs))
+
+	for _, t := range txs {
+		wg.Add(1)
+		go client.GetTransactionReceipt(receiptsChan, t.Meta.TxID)
```

Comment:
This could open a lot of connections, consider HTTP/1.1 pipelining or HTTP/2 multiplexing as a more efficient alternative

Reply:
This code was missed during rebase, added close channel here https://github.com/TrustWallet/blockatlas/pull/70/commits/9202343889007b69ca4b882a786aae6601794960. It's great idea to use `fasthttp` lib and pipelining, but for let's complete all planned chains and next step we can use your proposed improvements.

Reply:
Agreed, let's postpone it.